src/site/content/en/blog/async-functions/index.md

```diff
@@ -0,0 +1,392 @@
+
+Although you're writing code that looks synchronous, ensure you don't miss the
+opportunity to do things in parallel.
+
+```js
+async function series() {
+  await wait(500); // Wait 500ms…
+  await wait(500); // …then wait another 500ms.
+  return "done!";
+}
+```
+
+The above takes 1000ms to complete, whereas:
+
+```js
+async function parallel() {
+  const wait1 = wait(500); // Start a 500ms timer asynchronously…
+  const wait2 = wait(500); // …meaning this timer happens in parallel.
+  await wait1; // Wait 500ms for the first timer…
+  await wait2; // …by which time this timer has already finished.
+  return "done!";
+}
+```
+
+The above takes 500ms to complete, because both waits happen at the same time.
+Let's look at a practical example.
+
+### Example: outputting fetches in order
+
+Say you wanted to fetch a series of URLs and log them as soon as possible, in the
+correct order.
+
+*Deep breath* - here's how that looks with promises:
+
+```js
+function logInOrder(urls) {
+  // fetch all the URLs
+  const textPromises = urls.map(url => {
+    return fetch(url).then(response => response.text());
+  });
+
+  // log them in order
+  textPromises.reduce((chain, textPromise) => {
+    return chain.then(() => textPromise)
+      .then(text => console.log(text));
+  }, Promise.resolve());
+}
+```
+
+Yeah, that's right, I'm using `reduce` to chain a sequence of promises. I'm *so
+smart*. But this is a bit of *so smart* coding you're better off without.
+
+However, when converting the above to an async function, it's tempting to go
+*too sequential*:
+
+<span class="compare-worse">Not recommended</span> - too sequential
+
+```js
+async function logInOrder(urls) {
+  for (const url of urls) {
+    const response = await fetch(url);
+    console.log(await response.text());
+  }
+}
+```
+
+Looks much neater, but my second fetch doesn't begin until my first fetch has
+been fully read, and so on. This is much slower than the promises example that
+performs the fetches in parallel. Thankfully there's an ideal middle-ground:
+
+<span class="compare-better">Recommended</span> - nice and parallel
+
+```js
+async function logInOrder(urls) {
+  // fetch all the URLs in parallel
+  const textPromises = urls.map(async url => {
+    const response = await fetch(url);
+    return response.text();
+  });
+
+  // log them in sequence
+  for (const textPromise of textPromises) {
+    console.log(await textPromise);
+  }
+}
+```
+
+In this example, the URLs are fetched and read in parallel, but the "smart"
+`reduce` bit is replaced with a standard, boring, readable for-loop.
+
+## Browser support and workarounds
+
+At the time of writing, async functions are enabled by default in Chrome, Edge,
+Firefox, and Safari.
+
+### Workaround: generators
+
+If you're targeting browsers that support generators (which includes
+[the latest version of every major browser](http://kangax.github.io/compat-table/es6/#test-generators){:.external}
```

Comment:
Netlify is failing to build because somewhere in this file, one of these `{: }` directives isn't working. Taking a quick glance I was wondering if it's this one because it doesn't have a space between the colon and the dot?

Reply:
I tried adding a space. See if that works.

Reply:
Everything seems to pass now. PTAL.